### DISCUSSION TITLE
Route constraint in OpenTelemetry.Instrumentation.AspNetCore

### DISCUSSION THREAD

**Discussion by dominikjeske** [open-telemetry/opentelemetry-dotnet#4432] [General]
My API controllers have route constraints - for example  "/api/sample/{channel:ib}/v{v:apiVersion}/[controller]/[action]" and because of this metric is not reporting actual path because it uses RoutePattern.RawText (https://github.com/open-telemetry/opentelemetry-dotnet/blob/ef0c9217087bfca0d6fe9206ccde343eb4e5738c/src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInMetricsListener.cs#L95) and not context.Request.Path.

Actual call:
 "/api/sample/ib/v1/controller/action" 

Mteric output:
 "/api/sample/{channel:ib}/v{v:apiVersion}/controller/action" 

Is this behavior expected?

**Comment by vishweshbankwar** [comment]
Yes - `http.route` is set as per the [specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/http-metrics.md#metric-httpserverduration)
